# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 46a5babaafd2
# Full Hash: 46a5babaafd2c0f2198f353d62254cf518dab391
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-15 04:16:11
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Implement initialization of entire persistent storage; r=dom-storage-reviewers,jari
#   
#   IndexedDB still calls QuotaManager::EnsurePersistentOriginIsInitializedInternal
#   from its idle database maintenance. However, we want to get rid of that so the
#   method can be made private. The idle database maintenance should only rely on
# ==============================================================================

diff -r 2ff5f3aaf841 -r 46a5babaafd2 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Mon Oct 14 10:00:26 2024 +0000
+++ b/dom/quota/QuotaManager.h	Mon Oct 14 10:06:14 2024 +0000
@@ -88,6 +88,7 @@
   friend class FinalizeOriginEvictionOp;
   friend class GroupInfo;
   friend class InitOp;
+  friend class InitializePersistentStorageOp;
   friend class InitTemporaryStorageOp;
   friend class OriginInfo;
   friend class ShutdownStorageOp;
@@ -363,6 +364,29 @@
   nsresult EnsureStorageIsInitializedInternal();
 
  public:
+  RefPtr<BoolPromise> InitializePersistentStorage();
+
+  RefPtr<BoolPromise> InitializePersistentStorage(
+      RefPtr<UniversalDirectoryLock> aDirectoryLock);
+
+  RefPtr<BoolPromise> PersistentStorageInitialized();
+
+  bool IsPersistentStorageInitialized() const {
+    AssertIsOnOwningThread();
+
+    return mPersistentStorageInitialized;
+  }
+
+  bool IsPersistentStorageInitializedInternal() const {
+    AssertIsOnIOThread();
+
+    return mPersistentStorageInitializedInternal;
+  }
+
+ private:
+  nsresult EnsurePersistentStorageIsInitializedInternal();
+
+ public:
   RefPtr<BoolPromise> InitializePersistentOrigin(
       const PrincipalInfo& aPrincipalInfo);
 
@@ -884,6 +908,8 @@
   uint64_t mTemporaryStorageUsage;
   int64_t mNextDirectoryLockId;
   bool mStorageInitialized;
+  bool mPersistentStorageInitialized;
+  bool mPersistentStorageInitializedInternal;
   bool mTemporaryStorageInitialized;
   bool mTemporaryStorageInitializedInternal;
   bool mCacheUsable;